### PR TITLE
General validation and cleaning

### LINKED ISSUE #23: Solve remove all remaining TODOs from code

### LINKED ISSUE #24: Validate all positive inputs
we should review and validate all the entrypoint inputs that should be positive

### LINKED ISSUE #31: Pair meta_info is hardcoded
a decision should be made regarding  hardcoded `name` and `symbol` of Pair's implementation

### PR DESCRIPTION
**Pull request by bogdan-manole**
Closes #31
Closes #23 
Closes #24

### CODE REVIEW

File: contracts/AedexV2Factory.aes [hunk 1]
```diff
@@ -39,8 +39,6 @@ contract AedexV2Factory =
             , allow_debug_mode = Option.default(false,allow_debug_mode)
             }
 
-    //TODO: this is a hack to store it's own address into state as a IAedexV2Factory
-    //try finding something more suitable
     stateful entrypoint set_this_factory( this_factory: IAedexV2FactoryForPair ) =
         require(Contract.address == this_factory.address,"AedexV2Factory: NOT_SAME_FACTORY")
         put( state { this_factory = Some(this_factory) })
```
Comment (kenodressel) [inline contracts/AedexV2Factory.aes:44]: are we happy with this? if so I am fine with removing this todo but I can't find a paper trail of the discussion?

**Comment by kenodressel** [comment]
Looks good otherwise :+1:

File: contracts/AedexV2Factory.aes (hunk 1 above)
Comment (bogdan-manole) [reply contracts/AedexV2Factory.aes:44]: Didn't find any other solution to pass the own address casted to some interface to a remote call. Indeed I was asking on the WIRE, but here is a good place to talk about it.

Comment (kenodressel) [reply contracts/AedexV2Factory.aes:44]: So we can not set it during init and Contract.address is not available where we need it? Where would that be?

Comment (kenodressel) [reply contracts/AedexV2Factory.aes:44]: I just saw its only the cloning. Contract.address is not an option there? I remeber we had this discussion already but i forgot the details...

Comment (bogdan-manole) [reply contracts/AedexV2Factory.aes:44]: If I would be able to cast `Contract.address` to  `IAedexV2FactoryForPair` that would be great, but the Contract.address is of type `address`, so compiler complains if I pass it like that. 
The only option to fool the compiler was to receive it through a regular entrypoint as a `IAedexV2FactoryForPair` parameter and from 'ouside world' to give the address of the actual factory.

Comment (marc0olo) [reply contracts/AedexV2Factory.aes:44]: what was the response of the core devs? did we raise that topic there?

Comment (bogdan-manole) [reply contracts/AedexV2Factory.aes:44]: I was asking on the WIRE not on slack, nobody responded me with a alternative for this.

Comment (marc0olo) [reply contracts/AedexV2Factory.aes:44]: then we should probably tag @hanssv here and ask him =)

**Review by thepiwo — APPROVED** [review]
looks good

File: contracts/AedexV2Factory.aes (hunk 1 above)
Comment (bogdan-manole) [reply contracts/AedexV2Factory.aes:44]: actually i found [this](https://github.com/aeternity/aesophia/blob/master/docs/sophia_stdlib.md#to_contract), i will try it and come back with news

Comment (bogdan-manole) [reply contracts/AedexV2Factory.aes]: done it, made the modifications

**Pull request merged**